include/realizations/catchment/Catchment_Formulation.hpp

```diff
@@ -14,9 +14,17 @@ namespace realization {
     class Catchment_Formulation : public Formulation, public HY_CatchmentArea, public Et_Accountable {
         public:
             Catchment_Formulation(std::string id, std::shared_ptr<data_access::GenericDataProvider> forcing, utils::StreamHandler output_stream)
-                : Formulation(id), HY_CatchmentArea(forcing, output_stream) { };
-
-            Catchment_Formulation(std::string id) : Formulation(id){};
+                : Formulation(id), HY_CatchmentArea(forcing, output_stream) { 
+                    // Assume the catchment ID is equal to or embedded in the formulation `id`
+                    size_t idx = id.find(".");
+                    set_catchment_id( idx == std::string::npos ? id : id.substr(0, idx) );
+                };
+
+            Catchment_Formulation(std::string id) : Formulation(id){
+                    // Assume the catchment ID is equal to or embedded in the formulation `id`
+                    size_t idx = id.find(".");
+                    set_catchment_id( idx == std::string::npos ? id : id.substr(0, idx) );
+            };
```

Comment:
Yeah, this bugged me a little... I'm willing to duplicate something (in this length/complexity neighborhood) twice but not three times. I did _not_ want to put it in the getter... arguably, it shouldn't be in the setter because that should receive something that is already a _catchment_ ID, while this is a _formulation_ ID... this code (functionally speaking) translates a formulation ID  into a catchment ID. So... yeah short of breaking it into a utility function (probably overkill for something this small) I don't see a great answer.